waspc/src/Wasp/Node/Version.hs

```diff
@@ -39,79 +40,101 @@ data VersionCheckResult
+  result <- getInstalledVersion
+  return $ case result of
+    Left errorMsg -> VersionCheckFail errorMsg
+    Right userVersion ->
+      if SV.isVersionInRange userVersion $ SV.Range [SV.gte oldestSupportedVersion]
+        then VersionCheckSuccess
+        else VersionCheckFail $ versionMismatchErrorMessage userVersion
+  where
+    versionMismatchErrorMessage :: SV.Version -> ErrorMessage
+    versionMismatchErrorMessage userVersion =
+      unlines
+        [ "Your " ++ commandName ++ " version does not meet Wasp's requirements! You are running " ++ commandName ++ " " <> show userVersion <> ".",
+          "Wasp requires " ++ commandName ++ " version " <> show oldestSupportedVersion <> " or higher."
+        ]
+
+parseVersionFromCommandOutput :: String -> [String] -> IO (Either ErrorMessage SV.Version)
+parseVersionFromCommandOutput commandName args = do
+  result <-
+    catchIOError
+      (Right <$> P.readProcessWithExitCode commandName args "")
+      ( \e ->
+          if isDoesNotExistError e
+            then return $ Left commandNotFoundErrorMessage
+            else return $ Left $ unkownErrorErrorMessage e
+      )
+  return $ case result of
+    Left processError ->
+      Left $ runningCommandFailedErrorMessage processError
+    Right (ExitFailure exitCode, _, stderr) ->
+      Left $ commandFailedErrorMessage exitCode stderr
     Right (ExitSuccess, stdout, _) ->
-      parseNodeVersionOutput stdout
-        & left
-          ( \e ->
-              "Wasp failed to the parse `node` version provided by `node --version`.\n"
-                <> (show e <> "\n")
-                <> "This is most likely a bug in Wasp, please file an issue."
-          )
-
--- | Extracts node version from the output of `node --version`.
-parseNodeVersionOutput :: String -> Either P.ParseError SV.Version
-parseNodeVersionOutput = P.parse nodeVersionParser ""
+      case parseVersionFromOutput stdout of
+        Left parseError -> Left $ parseVersionFailedErrorMessage parseError
+        Right version -> Right version
   where
-    nodeVersionParser = skipAnyCharTillMatch SV.versionParser
-    skipAnyCharTillMatch p = P.manyTill P.anyChar (P.lookAhead $ P.try p) >> p
-
-nodeNotFoundMessage :: String
-nodeNotFoundMessage = "`node` command not found!"
-
-makeNodeUnknownErrorMessage :: IOError -> String
-makeNodeUnknownErrorMessage err =
-  unlines
-    [ "An unknown error occured while trying to run `node --version`:",
-      indent 2 $ show err
-    ]
+    commandWithArgs = unwords $ commandName : args
+
+    commandNotFoundErrorMessage :: ErrorMessage
+    commandNotFoundErrorMessage = "`" ++ commandWithArgs ++ "` command not found!"
+
+    unkownErrorErrorMessage :: IOError -> ErrorMessage
+    unkownErrorErrorMessage err =
+      unlines
+        [ "An unknown error occurred while trying to run `" ++ commandWithArgs ++ ":",
+          indent 2 $ show err
+        ]
+
+    runningCommandFailedErrorMessage :: String -> ErrorMessage
```

Comment:
This sounds as if "running command" failed. In the sense that command was running, and then it failed. But we awnt to say that we failed to run it. So it should be `failedToRunCommandErrorMessage`.